src/mmj/lang/Formula.java

```diff
@@ -610,9 +610,24 @@ public String toString() {
      * @return String for the expression portion of Formula
      */
     public String exprToString() {
-        final StringBuilder sb = new StringBuilder(sym.length * 3);
+        return exprToString(1, sym.length);
+    }
+
+    /**
+     * Computes a character string version of the formula.
+     * <p>
+     *
+     * @return String for the expression portion of Formula
+     */
+    private String exprToString(int symbolStart, int symbolCount) {
+        final StringBuilder sb = new StringBuilder(symbolCount * 3);
+        if (symbolStart <= 0) {
+            // include "type code".
+            sb.append(sym[0].getId());
+            symbolStart = 1;
+        }
```

Comment:
The test for 0 is essential. For negatives, it would be better to put an assert here.